### PR TITLE
Issue 3804 application fails to start 610 pre14 snapshot with beandefinitionoverrideexception invalid bean definition with name loadidsstep

### PR DESCRIPTION
**Pull request by jmarchionatto**

### CODE REVIEW

File: hapi-fhir-storage-batch2-jobs/src/main/java/ca/uhn/fhir/batch2/jobs/config/SharedCtx.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class SharedCtx {
+
+	@Bean
+	public LoadIdsStep loadIdsStep(IBatch2DaoSvc theBatch2DaoSvc) {
+		return new LoadIdsStep(theBatch2DaoSvc);
+	}
```
Comment (tadgh) [inline hapi-fhir-storage-batch2-jobs/src/main/java/ca/uhn/fhir/batch2/jobs/config/SharedCtx.java:14]: Suggestion: Maybe rename class to `SharedBatchCtx` for clarity?

File: hapi-fhir-storage-batch2-jobs/src/main/java/ca/uhn/fhir/batch2/jobs/expunge/DeleteExpungeAppCtx.java [hunk 2]
```diff
@@ -36,7 +36,7 @@
 import org.springframework.context.annotation.Configuration;
 
 @Configuration
-public class DeleteExpungeAppCtx {
+public class DeleteExpungeAppCtx extends SharedCtx {
```
Comment (tadgh) [inline hapi-fhir-storage-batch2-jobs/src/main/java/ca/uhn/fhir/batch2/jobs/expunge/DeleteExpungeAppCtx.java:39]: Question: Why not just @Import the ctx instead? Spring should auto-resolve multiple @Imports.

Comment (tadgh) [reply hapi-fhir-storage-batch2-jobs/src/main/java/ca/uhn/fhir/batch2/jobs/expunge/DeleteExpungeAppCtx.java:39]: If memory serve spring does _really_ poorly with inherited configs.

**Review by tadgh — APPROVED** [review]
Approved pending addressed comments

File: hapi-fhir-storage-batch2-jobs/src/main/java/ca/uhn/fhir/batch2/jobs/config/SharedCtx.java (hunk 1 above)
Comment (jmarchionatto) [reply hapi-fhir-storage-batch2-jobs/src/main/java/ca/uhn/fhir/batch2/jobs/config/SharedCtx.java:14]: Good idea. Done

**Pull request merged**